miden/tests/integration/exec_iters.rs

```diff
@@ -20,90 +20,103 @@ fn test_exec_iter() {
         VmState {
             clk: 0,
             op: None,
+            asmop: None,
             stack: [16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1].to_elements(),
             fmp,
             memory: Vec::new(),
         },
         VmState {
             clk: 1,
             op: Some(Operation::Span),
+            asmop: None,
             stack: [16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 1].to_elements(),
             fmp,
             memory: Vec::new(),
         },
         VmState {
             clk: 2,
             op: Some(Operation::Push(Felt::new(1))),
+            asmop: Some(AsmOpInfo::new("popw.mem.1".to_string(), 6)),
             stack: [1, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2].to_elements(),
             fmp,
             memory: Vec::new(),
         },
         VmState {
             clk: 3,
             op: Some(Operation::MStoreW),
+            asmop: None,
```

Comment:
One way would be to change `AsmOpInfo` struct to have this information. It could have a field called `pos`/`position` or `loc`/`location` with location of the AsmOp in the sequence of operation. This could be 1-indexed to remove ambiguity.
This would look like `AsmOpInfo("pop.local", 1, 10)` ... `AsmOpInfo("pop.local", 10, 10)`. This would map all operations in the sequence to a decorator and is probably a bit redundant. Another option is to add it to only VmState by changing the decorator the `asmop` field to another custom type that has this field. I think probably the most favorable way would be to only include this information at the time of displaying it so maybe add it to display trait of VmState?

Reply:
I like the option of adding another custom type. In that type, we could have:
* The original full name of the instruction.
* Number of cycles it took to execute.
* Index of the current operation within the instruction.

So, it would read something like this:
```
"pop.local.0", 10, 1   // pop.local 1 out of 10
"pop.local.0", 10, 2   // pop.local 2 out of 10
...
"pop.local.0", 10, 10  // pop.local 10 out of 10
```

In this type, we could also add a function to return the generalized name of the instruction (i.e., "push" instead of "push.123"). By the way, I'm not sure removing the last int parameter works in all cases. For example, it might be good to know which `movup` or `movdn` was used. So, we should probably have a more custom function for mapping instructions to their generalized form - and this could be that function.

Reply:
I've changed `total vm cycles` per assembly instruction in latest commit, but I think it might be counter intuitive. Consider the following program:
`proc.foo.1 pop.local.0 end begin popw.mem.1 push.17 push.1 movdn.2 exec.foo end`
and the corresponding AsmOpStats:
```
vec![
    AsmOpStats::new("movdn.2".to_string(), 1, 1, 1),
    AsmOpStats::new("pop.local".to_string(), 10, 1, 10),
    AsmOpStats::new("popw.mem".to_string(), 6, 1, 6),
    AsmOpStats::new("push".to_string(), 1, 2, 3),
]
``` 
Here `push.1` takes 2 cycles and `push.17` takes 1 cycle. If the order of push.17 and push.1 changed in the above program, the corresponding AsmOpStats would change to `AsmOpStats::new("push".to_string(), 2, 2, 3)` indicating 2 cycles per instruction occurrence, with frequency of 2 and total vm cycles per instruction = 3.

Reply:
There are a couple of ways to address this:

1. We could group operations with the same number of cycles. For example, "push" operation would be split into a few groups: "push.0", "push.x", "push.x.x" etc.
2. Instead of number of cycles, we could have "average number of cycles". This would be a float computed as total VM cycles / frequency. This doesn't even need to be a separate field - could be just computed dynamically.

I think the first approach is probably more informative, but also more complex. Maybe we start with the second approach and move to the first one in future PRs?

Reply:
Makes sense. Removed `num_cycles` as a field and added `Avg instruction cycles` to program info's display trait.